community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -408,9 +410,16 @@ public void start() throws IOException
 
         life.add( recoveryCleanupWorkCollector );
 
-        schemaIndexProvider = dependencyResolver.resolveDependency( SchemaIndexProvider.class,
-                HighestSelectionStrategy.getInstance() );
-        dependencies.satisfyDependency( schemaIndexProvider );
+        AllByPrioritySelectionStrategy<SchemaIndexProvider> indexProviderSelection = new AllByPrioritySelectionStrategy<>();
+        SchemaIndexProvider defaultIndexProvider =
+                dependencyResolver.resolveDependency( SchemaIndexProvider.class, indexProviderSelection );
+
+        // todo do we really need to have this dependency?
```

Comment:
It was left there because that's how it was previously. Now the `SchemaIndexProviderMap` is used instead. It's not strictly required for us to remove this right now, since it may require some time and testing. We've created a card for it now.